src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -255,7 +255,10 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
       } else {
         loadConfig.setDestinationTable(tableRef);
 
-        if (allowSchemaRelaxation) {
+        //Schema update options should only be specified with WRITE_APPEND disposition,
+        // or with WRITE_TRUNCATE disposition on a table partition - The logic below should change when we support
+        // insertion into single partition
+        if (allowSchemaRelaxation && !JobInfo.WriteDisposition.WRITE_TRUNCATE.toString().equals(writeDisposition)) {
```

Comment:
it's usually better to change the string into the Enum and compare enums instead of changing enum to string and comparing.

Reply:
Done